routers/web/repo/shared/license.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package shared
+
+import (
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/services/context"
+	repo_service "code.gitea.io/gitea/services/repository"
+)
+
+func PrepareForLicenses(ctx *context.Context) bool {
+	repoLicenses, err := repo_model.GetRepoLicenses(ctx, ctx.Repo.Repository)
+	if err != nil {
+		ctx.ServerError("GetRepoLicenses", err)
+		return false
+	}
+	ctx.Data["DetectedRepoLicenses"] = repoLicenses.StringList()
+	ctx.Data["LicenseFileName"] = repo_service.LicenseFileName
```

Comment:
I know, this code was simply moved.
But… why are we storing request-specific data inside a global variable?

Reply:
I cannot answer the question. I guess it may allow a different license file name in the future. Maybe @yp05327 can answer.

Reply:
`LicenseFileName` in context data is used for the link.
![image](https://github.com/user-attachments/assets/7f240414-0d42-4b19-97c4-c39956b133ac)
At first, I reused the code of detecting `README` file, so the license file name is not a specific value, but according to the review of https://github.com/go-gitea/gitea/pull/24872#discussion_r1776315177 and https://github.com/go-gitea/gitea/pull/24872#discussion_r1776316043, I changed it into a specific value.
But I think is not enough, e.g. `LICENSES` or `license` is not supported, and if it is necessary, we can improve it in the future.